### ISSUE TITLE
build/hello-world: conflicting lockfiles breaks build

### ISSUE THREAD

**Issue by phillipj** [GoogleCloudPlatform/nodejs-getting-started#129]
Hi!

First of all, thanks a lot for great starter projects to quickly get something up-n-running 👍 

I started playing with App Engine yesterday and went through the tutorial of setting up the hello-world example of this repo. Sadly it crashed when trying to deploy it because there's two lockfiles present in the hello-world project.

```bash
$ gcloud app deploy --project phillips-playground

..

If this is your first deployment, this may take a while...done.

---------------------------------------------------------- REMOTE BUILD OUTPUT ----------------------------------------------------------
starting build "36e38007-e98a-4336-85c0-5bf3d75aee8c"

..

Starting Step #0
Step #0: Checking for Node.js.
Step #0: Application detection failed: Error: Cannot determine which package manager to use as both yarn.lock and package-lock.json files
 were detected.  The presence of yarn.lock indicates that yarn should be used, but the presence of package-lock.json indicates npm should
 be used.  Use the skip_files section of app.yaml to ignore the appropriate file to indicate which package manager to use.
Finished Step #0
ERROR
```

I assume the most ideal would be if the build & deploy step would succeed right away, instead of making the developer see this fail and then fix a couple of things, before being able to try again?

If so, I'd suggest two different approaches:

1. List one of the lockfiles under `skip_files` section of `app.yaml`
2. Delete one of the lockfiles

Personally leaning towards (2) as ignore one of the lockfiles begs the question why we should have that file in the project in the first place..

Any thoughts? 

P.S. I'm more than willing to open a PR with needed changes, I just wanted to hear your thoughts by opening an issue first.

**Comment by Zev-Chonoles** [comment]
I was just going through the node tutorial, and I experienced this same problem:
<pre>
---------------------------------------------------------------------------- REMOTE BUILD OUTPUT -----------------------------------------------------------------------------
starting build "c4b0677e-2862-44ef-bce1-4fd616027378"

FETCHSOURCE
Fetching storage object: gs://staging.mytest-187905.appspot.com/us.gcr.io/mytest-187905/appengine/default.20171202t233711:latest#1512279434342879
Copying gs://staging.mytest-187905.appspot.com/us.gcr.io/mytest-187905/appengine/default.20171202t233711:latest#1512279434342879...
/ [1 files][ 77.8 KiB/ 77.8 KiB]
Operation completed over 1 objects/77.8 KiB.
BUILD
Step #0: Pulling image: gcr.io/gcp-runtimes/nodejs/gen-dockerfile@sha256:196bc20ff8d91905dc071100399538814e2c619d0d27576c35a6405674da696c
Step #0: sha256:196bc20ff8d91905dc071100399538814e2c619d0d27576c35a6405674da696c: Pulling from gcp-runtimes/nodejs/gen-dockerfile
Step #0: Digest: sha256:196bc20ff8d91905dc071100399538814e2c619d0d27576c35a6405674da696c
Step #0: Status: Downloaded newer image for gcr.io/gcp-runtimes/nodejs/gen-dockerfile@sha256:196bc20ff8d91905dc071100399538814e2c619d0d27576c35a6405674da696c
Starting Step #0
Step #0: Checking for Node.js.
Step #0: Application detection failed: Error: Cannot determine which package manager to use as both yarn.lock and package-lock.json files were detected.  The presence of yarn.lock indicates that yarn should be used, but the presence of package-lock.json indicates npm should be used.  Use the skip_files section of app.yaml to ignore the appropriate file to indicate which package manager to use.
Finished Step #0
ERROR
ERROR: build step "gcr.io/gcp-runtimes/nodejs/gen-dockerfile@sha256:196bc20ff8d91905dc071100399538814e2c619d0d27576c35a6405674da696c" failed: exit status 1
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

ERROR: (gcloud.app.deploy) Cloud build failed. Check logs at https://console.cloud.google.com/gcr/builds/c4b0677e-2862-44ef-bce1-4fd616027378?project=mytest-187905 Failure status: UNKNOWN: Error Response: [2] Build failed; check build logs for details
</pre>
I'm not sure what the right step is, but if someone could take a lot at this again, it'd be very helpful.

**Comment by phillipj** [comment]
Delete or ignore the lockfile you don't about.

If you're a traditional npm user, delete/ignore yarn.lock, or else if
you're a yarn user, delete/ignore package-lock.json.

**Comment by ace-n** [comment]
Thanks @phillipj for reporting this - it (and many other long-standing bugs) have since been fixed by #124.

Please feel free to reopen this issue if you continue having problems with this. Thanks!

**Comment by phillipj** [comment]
Thanks for following up @ace-n and congrats on merging that enormous list of fixes 😄
